Modules/_cursesmodule.c

```diff
@@ -2040,43 +2056,59 @@ PyCursesWindow_InStr(PyObject *op, PyObject *args)
     PyCursesWindowObject *self = _PyCursesWindowObject_CAST(op);
 
     int x, y, n;
-    char rtn[1024]; /* This should be big enough.. I hope */
-    int rtn2;
+    int rtn;
+
+    /* could make the buffer size larger/dynamic */
+    Py_ssize_t max_buf_size = 2048;
+    PyObject *result = PyBytes_FromStringAndSize(NULL, max_buf_size);
+    char *buf = PyBytes_AS_STRING(result);
```

Comment:
```suggestion
    PyObject *result = PyBytes_FromStringAndSize(NULL, max_buf_size);
    if (result == NULL) {
        return NULL;
    }
    char *buf = PyBytes_AS_STRING(result);
```

and ditto for the other function (I forgot about this).

Reply:
should we return a PyErr_NoMemory() in this case?

Reply:
No, an exception should already be set by `PyBytes_FromStringAndSize`